backend/src/apiserver/template/tekton_template.go

```diff
@@ -322,8 +322,24 @@ func (t *Tekton) injectArchivalStep(workflow util.Workflow, artifactItemsJSON ma
 					t.getEnvVar("TRACK_ARTIFACTS", strconv.FormatBool(trackArtifacts)),
 					t.getEnvVar("STRIP_EOF", strconv.FormatBool(stripEOF)),
 				)
+				container.Command = []string{"sh", "-c"}
+				container.Args = []string{artifactScript}
+
+				step := workflowapi.Step{Container: container}
+
+				if task.TaskSpec.Results != nil && len(task.TaskSpec.Results) > 0 {
+					// move all results to /tekton/home/tep-results to avoid result duplication in copy-artifacts step
+					// TODO: this causes eof strip fails, since no results under /tekton/results after this step
```

Comment:
can you add if multiple copy steps exist, don't run the eof command injection? Thanks